src/backend/ApiTypes.ts

```diff
@@ -0,0 +1,90 @@
+export interface DirectoryEntry {
+    name?: string,
+    type?: EntryType,
+    mode?: Permissions,
+    size?: number,
+    mtime?: Time,
+    uid?: number,
+    gid?: number,
+    obj?: ObjectID,
+    summ?: DirectorySummary
+}
+
+export interface DirectorySummary {
+    size: number,
+    files: number,
+    symlinks: number,
+    dirs: number,
+    maxTime: Date,
+    incomplete?: string,
+    numFailed: number,
+    numIgnoredErrors?: number,
+    errors?: EntryWithError[]
+}
+
+export interface EntryWithError {
+    path: string,
+    error: string
+}
+
+export type Time = string | number | Date;
+
+export type EntryType = string;
+
+export interface Obj {
+
+}
+
+export interface InfoStruct {
+    contentID: ObjectID,
+    packFile?: string,
+    time: string,
+    oritinalLength: number,
+    packedLength: number,
+    packOffset?: number,
+    deleted: boolean,
+    formatVersion: number,
+    compression?: HeaderID,
+    encryptionKeyID?: number
+}
+
+export enum HeaderID {
```

Comment:
can we just use a number here? This enum does not belong in the UI (yet, hopefully ever)

Reply:
I've changed it to be a number (without description) but I retained that it can only be this value, or that value, ...; i.e. I left all of the possible values in, but removed the descriptions/names in the enum.